### PR TITLE
Topic0

### PR DESCRIPTION
**Pull request by natapg**
Topic 0 point 1 added and Topic 1

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (EduardoUTN) [inline pom.xml]: This file is Topic1, right?
Try to read how to skip test:
https://www.mkyong.com/maven/how-to-skip-maven-unit-test/

File: src/main/java/com/natapg/App.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.natapg;
+
+import com.natapg.db.MockedConnection;
+
+public class App
+{
```
Comment (LeoBur) [inline src/main/java/com/natapg/App.java]: We use the google code style 

> No line break before the opening brace.

https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style

File: Topic0/Point3/point3/ConnectDB.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.natapg.point3;
+
+import com.natapg.inter.DBConecction;
+
+public class ConnectDB implements DBConecction {
+
+  @Override
+  public boolean connect() {
+    System.out.println("Connecting to the database...");
```
Comment (sebastianscatularo) [inline Topic0/Point3/point3/ConnectDB.java]: avoid the use of System.out.println(...)

File: Topic0/Point3/point3/ProxyConnection.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    } else {
+      return realConnection.connect();
+    }
+  }
+
+  @Override
+  public void close() {
+    if (realConnection == null) {
+      realConnection = new ConnectDB();
+      realConnection.close();
```
Comment (EduardoUTN) [inline Topic0/Point3/point3/ProxyConnection.java]: If `realConnection is null` you are creating an Object and later closing the same.
Instead doing this you can avoid the Null Pointer Exception (NPE) with a try/catch Block

File: Topic0/Point1/src/main/java/com/natapg/point1/MockedConnection.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ */
+public class MockedConnection {
+
+  private static MockedConnection connection = new MockedConnection();
+
+  private MockedConnection() {
+  }
+
+  public static MockedConnection getInstance() {
+    if (connection == null) {
```
[the remark this answers is not in the document]
Comment (EduardoUTN) [reply Topic0/Point1/src/main/java/com/natapg/point1/MockedConnection.java]: Check this line and the line 8. If you instantiate this class, when connection is `null`?.

**Pull request merged**